File: .claude/hooks/load-git-context.mjs
#!/usr/bin/env node
// @hookstack session-start-load-git-context
// Injecte le contexte git (branche, statut) dans chaque prompt (UserPromptSubmit)
import { execSync } from 'child_process';
import { fileURLToPath } from 'url';

function defaultExec(cmd) {
  try { return execSync(cmd, { encoding: 'utf8', timeout: 5_000 }).trim(); } catch { return ''; }
}

export function run({ exec = defaultExec } = {}) {
  const branch = exec('git branch --show-current') || exec('git rev-parse --abbrev-ref HEAD');
  const commit = exec('git log -1 --pretty="%h %s"');
  const status = exec('git status --short');

  if (!branch) return null;

  const lines = ['## Contexte Git', `- Branche : \`${branch}\``];
  if (commit) lines.push(`- Dernier commit : ${commit}`);
  if (status) lines.push(`- Fichiers modifiés :\n${status.split('\n').map((l) => `  ${l}`).join('\n')}`);
  else lines.push('- Répertoire de travail propre');

  return lines.join('\n') + '\n';
}

/* v8 ignore next 4 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
  const result = run();
  if (result) process.stdout.write(result);
}

File: .claude/hooks/load-git-context.py
#!/usr/bin/env python3
# @hookstack session-start-load-git-context
"""Injects git context (branch, status) at session start (SessionStart)."""
import subprocess
import sys


def _exec(cmd):
    try:
        return subprocess.run(
            cmd, shell=True, capture_output=True, text=True, timeout=5, check=True
        ).stdout.strip()
    except Exception:
        return ""


def run(*, exec_cmd=None):
    if exec_cmd is None:
        exec_cmd = _exec
    branch = exec_cmd("git branch --show-current") or exec_cmd(
        "git rev-parse --abbrev-ref HEAD"
    )
    commit = exec_cmd('git log -1 --pretty="%h %s"')
    status = exec_cmd("git status --short")

    if not branch:
        return None

    lines = ["## Contexte Git", f"- Branche : `{branch}`"]
    if commit:
        lines.append(f"- Dernier commit : {commit}")
    if status:
        lines.append(
            "- Fichiers modifiés :\n" + "\n".join(f"  {l}" for l in status.split("\n"))
        )
    else:
        lines.append("- Répertoire de travail propre")

    return "\n".join(lines) + "\n"


if __name__ == "__main__":
    result = run()
    if result:
        sys.stdout.write(result)
